helper/cert/source_gen.go

```diff
@@ -236,25 +236,31 @@ func (s *GenSource) retryUpdateSecret(ctx context.Context, leaderCh chan bool, b
 }
 
 func (s *GenSource) updateSecret(ctx context.Context, bundle Bundle) error {
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: certSecretName,
-		},
-		Data: map[string][]byte{
-			"cert": bundle.Cert,
-			"key":  bundle.Key,
-		},
-	}
-	// Attempt updating the Secret first, and if it doesn't exist, fallback to
-	// create
-	_, err := s.K8sClient.CoreV1().Secrets(s.Namespace).Update(ctx, secret, metav1.UpdateOptions{})
+	secret, err := s.K8sClient.CoreV1().Secrets(s.Namespace).Get(ctx, certSecretName, metav1.GetOptions{})
```

Comment:
I think we may want to do the get from the informer cache here, instead of the regular k8s client. That should help avoid any rate limiting from the k8s API, even though this get shouldn't be firing that often. Something like:

```suggestion
	secret, err := s.SecretsCache.Lister().Secrets(s.Namespace).Get(certSecretName)
```

Similar to getBundleFromSecret(): https://github.com/hashicorp/vault-k8s/blob/0603c2a03300773fd6ba6174c6d4327d70889d58/helper/cert/source_gen.go#L266-L269

Reply:
It's done 👍